Adding google analytics

Zayen-X opened the pull request:
**What it do**:
- Adds the gtag script to the main page

Improvement: use `react-ga` to hook directly into the router.

Zayen-X committed:
adding gtag to index.html

appleseed-iii committed:
1. add react-ga4 (react-ga does not work with new Google Analytics 4)
2. add history listener to Root.jsx via createBrowserHistory()
3. Remove BrowserRouter basename=# bc it breaks GA4 pageview tracking (I tested this. basename=# made nothing trackable)
4. Add Temp Alchemy Id 2

appleseed-iii committed:
Merge branch 'provider-call-debug-ii' into google-analytics

appleseed-iii committed:
remove tag manager from index.html

Zayen-X commented on src/hooks/web3Context.tsx:
```diff
@@ -35,20 +40,22 @@ function getAlchemyAPI(chainID: Number) {
+  // There is no lightweight way to test each URL. so just return a random one.
+  // if (workingURI !== undefined || workingURI !== "") return workingURI as string;
+  const randomIndex = Math.floor(Math.random() * allURIs.length);
+  return allURIs[randomIndex];
```
idk why, reading this uncomment block took so long. :hurtrealbad:

appleseed-iii replied:
just reworking the tempAPIkeys to use the same format as the permanent API Keys. All we really have to do now to toggle back & forth is to change `ALL_URIs`

Zayen-X commented on src/Root.jsx:
```diff
@@ -7,6 +7,18 @@ import { Web3ContextProvider } from "./hooks/web3Context";
+//
+// Initialize google analytics page view tracking
+import { createBrowserHistory } from "history";
+const gaHistory = createBrowserHistory();
```
Do we not need to instantiate `Router history=history`? 
This just works?

appleseed-iii replied:
it just works. I tested with my own GA4 tracking-id.

appleseed-iii replied:
narrator: it didn't just work. https://github.com/OlympusDAO/olympus-frontend/pull/372/commits/5dd6ad555a9a85dcafd75d7e813322bff39bb953 moved the code into `src/hooks/analytics` with a custom hook to make it all work properly.

Zayen-X commented on src/Root.jsx:
```diff
@@ -16,7 +28,7 @@ export default class Root extends Component {
       <Web3ContextProvider>
         <Provider store={store}>
-          <BrowserRouter basename={"/#"}>
+          <BrowserRouter>
```
What happens for people who have it bookmarked with the `/#` can we add redirects for now?

appleseed-iii replied:
ok, https://github.com/OlympusDAO/olympus-frontend/pull/372/commits/5dd6ad555a9a85dcafd75d7e813322bff39bb953, restores this functionality since it was needed for IPFS

appleseed-iii committed:
1. restore BrowserRouter basename for IPFS purposes
2. move react-ga into a custom hook, add that hook to App.jsx to fire on every location change
3. add location.hash into the GA4 path as a result of the basename={/#}

appleseed-iii committed:
fixing google analytics key

appleseed-iii commented on src/Root.jsx:
```diff
@@ -7,18 +7,6 @@ import { Web3ContextProvider } from "./hooks/web3Context";
-gaHistory.listen(location => {
-  ReactGA.send({ hitType: "pageview", page: location.pathname });
-});
-
```
none of the above works properly with `BrowserRouter`. And with `BrowserRouter basename={"/#"}` react-ga4 thinks that window.location.pathname is ALWAYS "/", which is why GA wasn't picking up the page views.

appleseed-iii committed:
combining analytics.js into useGoogleAnalytics.js

Zayen-X reviewed (commented):
LGTM. wee need GA. lol
